active-member-discord-bot/src/main/kotlin/regenerativeag/discord/client/ResetMembershipsClient.kt

```diff
@@ -0,0 +1,118 @@
+package regenerativeag.discord.client
+
+import kotlinx.coroutines.runBlocking
+import mu.KotlinLogging
+import regenerativeag.Database
+import regenerativeag.discord.ActiveMemberDiscordBot
+import regenerativeag.discord.Discord
+import regenerativeag.discord.model.Message
+import regenerativeag.discord.model.UserId
+import regenerativeag.model.ActiveMemberConfig
+import regenerativeag.model.MutablePostHistory
+import regenerativeag.model.PostHistory
+import java.time.LocalDate
+
+/** A DiscordClient which posts messages to appropriate rooms when adding/removing roles */
+class ResetMembershipsClient(
+    discord: Discord,
+    private val database: Database,
+    private val membershipRoleClient: MembershipRoleClient,
+    private val activeMemberConfig: ActiveMemberConfig,
+) : DiscordClient(discord) {
+    private val logger = KotlinLogging.logger { }
+
+    /** Read the message history, correct any issues in the DB, and correct any issues in people's roles */
+    fun cleanReload() {
+        logger.debug { "Reloading" }
+
+        val today = LocalDate.now()
+        val postHistory = fetchPostHistory(today)
+
+        logger.debug { "Overwriting post history: $postHistory" }
+        database.overwritePostHistory(postHistory)
+
+        updateMemberRolesGivenPostHistory(postHistory, today)
+
+        logger.debug { "Reload complete" }
+    }
+
+    fun updateMemberRolesGivenPostHistory(postHistory: PostHistory, today: LocalDate) {
+        // Iterate over every role, updating all members in that role
+        val computedMembersSets = ActiveMemberDiscordBot.computeActiveMembers(postHistory, today, activeMemberConfig)
+        val departedMemberIds = mutableSetOf<UserId>()
+        val previousMemberIds = mutableSetOf<UserId>()
+        computedMembersSets.zip(activeMemberConfig.roleConfigs).forEach { (members, roleConfig) ->
+            val result = updateRoleMembers(members, roleConfig)
+            departedMemberIds.addAll(result.departedMemberIds)
+            previousMemberIds.addAll(result.previousMemberIds)
```

Comment:
this was the major change in this file. Otherwise it was pretty much just a copy of what used to be called `ActiveMemberDiscordBot.Reloader`